google_cloud_automlops/orchestration/base.py

```diff
@@ -124,14 +167,15 @@ def _get_function_return_types(self) -> list:
         if not (hasattr(annotation,'__annotations__') and isinstance(annotation.__annotations__, dict)):
             raise TypeError(f'''Return type hint for function "{self.name}" must be a NamedTuple.''')
 
-        # Creates a dictionary of metadata for each object returned by component
+        # Creates a parameter object for each parameter returned by component
         outputs = []
```

Comment:
can we add a `hint` here so its more strongly typed?

`outputs = [] -> Parameter`

I am unsure if this works with Pydantic but wanted to see if it did then we should do this.

Reply:
No, Python doesn't allow specifications like that unless it's a parameter

Reply:
You could do something like `output: List[Parameter] = []`, per [this doc page](https://docs.python.org/3/reference/simple_stmts.html#annassign). 

Note that this will not throw an error at runtime, but it will throw a warning in your code editor if it's configured to do so, and if you use a tool like mypy, mypy will also complain if you pass the wrong data type. 

In general, type hints in Python are hints, not rules.